driven/storage/adapter.go

```diff
@@ -847,6 +847,38 @@ func (sa Adapter) DeleteMessageWithContext(ctx context.Context, orgID string, ap
 	return nil
 }
 
+// UpdateUnreadMessage updates a unread message in the recipients to read
+func (sa Adapter) UpdateUnreadMessage(message *model.Message, userID *string) (*model.Message, error) {
+	if message != nil {
+		updateRecipients := []model.Recipient{}
+		for _, recipient := range message.Recipients {
+			if *userID == *recipient.UserID {
+				if recipient.Read == false {
+					recipient.Read = true
+					updateRecipients = append(updateRecipients, recipient)
+					filter := bson.D{
+						primitive.E{Key: "org_id", Value: message.OrgID},
+						primitive.E{Key: "app_id", Value: message.AppID},
+						primitive.E{Key: "_id", Value: message.ID}}
+					update := bson.D{
+						primitive.E{Key: "$set", Value: bson.D{
+							primitive.E{Key: "recipients", Value: updateRecipients},
```

Comment:
This operation is supposed to update single recipient element in the recipients list - not the entire list!